src/warp/WarpRouteDeployer.ts

```diff
@@ -187,31 +195,40 @@ export class WarpRouteDeployer {
         `Fetching token metadata for ${base.address} on ${base.chainName}}`,
       );
       const provider = this.multiProvider.getProvider(base.chainName);
-      const erc20Contract = ERC20__factory.connect(base.address, provider);
-      const [name, symbol, decimals] = await Promise.all([
-        erc20Contract.name(),
-        erc20Contract.symbol(),
-        erc20Contract.decimals(),
-      ]);
-      return { name, symbol, decimals };
+      if (base.isNft) {
+        const erc721Contract = ERC721__factory.connect(base.address, provider);
+        const [name, symbol] = await Promise.all([
+          erc721Contract.name(),
+          erc721Contract.symbol(),
+        ]);
+        return { name, symbol, decimals: 0 };
+      } else {
+        const erc20Contract = ERC20__factory.connect(base.address, provider);
+        const [name, symbol, decimals] = await Promise.all([
+          erc20Contract.name(),
+          erc20Contract.symbol(),
+          erc20Contract.decimals(),
+        ]);
+        return { name, symbol, decimals };
+      }
     } else {
       throw new Error(`Unsupported token type: ${base}`);
     }
   }
 
   writeDeploymentResult(
-    contracts: HyperlaneContractsMap<HypERC20Factories>,
+    contracts: HyperlaneContractsMap<HypERC20Factories | HypERC721Factories>,
     configMap: ChainMap<TokenConfig & RouterConfig>,
     baseToken: Awaited<
-      ReturnType<typeof this.buildHypERC20Config>
+      ReturnType<typeof this.buildHypTokenConfig>
     >['baseToken'],
   ) {
     this.writeTokenDeploymentArtifacts(contracts, configMap);
     this.writeWarpUiTokenList(contracts, baseToken, configMap);
   }
 
   writeTokenDeploymentArtifacts(
-    contracts: HyperlaneContractsMap<HypERC20Factories>,
+    contracts: HyperlaneContractsMap<HypERC20Factories | HypERC721Factories>,
```

Comment:
Should we define this union type somewhere?

Reply:
Will do